x/aswap/codec.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package aswap;
+
+import "github.com/iov-one/weave/codec.proto";
+import "github.com/iov-one/weave/coin/codec.proto";
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+// Swap is designed to hold some coins for atomic swap, locked by preimage_hash
+message Swap {
+    // metadata is used for schema versioning support
+    weave.Metadata metadata = 1;
+    // sha256 hash of preimage, 32 bytes long
+    bytes preimage_hash = 2;
+    // src is a sender address
+    bytes src = 3 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+    // address is a swap address deduced from preimage_hash condition
+    bytes address = 4 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
```

Comment:
do we need this? I thought we can calculate from Swap itself.
but if helpful, it doesn't hurt